### DISCUSSION TITLE
Some thoughts on Vehicles

### DISCUSSION THREAD

**Discussion by lewcc** [ParadiseSS13/Paradise#20422] [Documents]
# Vehicle Thoughts, Distilled

This is something that's been on my mind for quite some time. Vehicles are in a bit of a weird state on Paradise. We have a few key vehicles that see pretty consistent use (janicart, ambulance, maybe the secway?), but that's about it. And they've generally been stagnant as far as balance goes. 

I think the ambulance is a good example to look at. Here's a rescue vehicle that's supposed to be the main tool of a paramedic. Cool, so it has a siren, and you can hook a bed to it. And it goes fast, right?

...right?

Well, no. Historically (and this is before my time, so somewhat anecdotally), vehicles were used and abused by powergamers on both sides of conflict. Antags could take them and run away without fearing shots, and seccies might comandeer a fast ambulance to chase down antags. The biggest reason: no damage slowdown. Vehicles are very binary: they either work or they're broken. When they work, they go at full speed, and are fully maneuverable, until they break (hell, can they even break?). If a vehicle moves at a faster speed than security, there is no good way to stop it. I'd have to check, but I don't think ions even do anything to them right now.

And this kind of sucks, because it means vehicles can't really have any advantages beyond "hey, look at me! I'm driving a VEHICLE! **I AM NOT FAIL RP!"**

I've been hearing a lot of talk, but haven't really seen too much change around vehicles, so I'm hoping to distill some of the ideas I've heard into a single document here. 

### Damage Failure Modes

One of the big problems I mentioned with vehicles is that their damage is really binary, and there isn't really any good support for failure modes. The easy way out is to give things an EMP weakness, but I think a much more interesting method would be to give vehicles some different modes of failure as they're damaged, much like how mechs and borgs make up for not having a damage slowdown by becoming worse and worse over time, possibly experiencing component failures.

Some examples I can think of might be:
* Tire failures
    * Running over something sharp with enough force (a knife, a bear trap) could puncture your tires, causing slowdown and a severe lack of control.
    * On the other hand, something like running over a bola could cause your wheels to temporarily lock up, giving folks just enough time to get to you.
    
* Internal component failure
    An engine or other component could blow, causing some sort of debuff. 
    * An engine could fail, for instance, causing severe slowdown.
    * A radiator or coolant system could fail, releasing large amounts of steam or smoke and blinding the user (also, providing a bit of a smokescreen).
    * Engine could also overheat after taking enough burn damage from lasers, causing significant slowdown.
        * This could also cause burn damage to the user.
    * For electrical vehicles (read: most), the user could take gradual burn damage from an electrical failure shocking them.
    
    
* Critical failures
    Just to up the ante a bit, some damage modes past a certain level could roll for some finishing effects.
    * Simple shutdown: Enough EMPs could cause this, having it fall apart underneath you and become defunct until repaired.
    * Critical overload: vehicle overheats enough and may explode violently.

...and more among those lines. I think it'd be cool if the door were left open.

## improving vehicle interactions

Currently, interacting with vehicles is pretty simple, with some more advanced features tacked on. You hop on and use a key, boom, you're driving. It just follows your movement, and your hands are free to do whatever.
I think it'd make sense to prevent you from just running and gunning in a vehicle. If you want to drive, you should have at least one hand dedicated to steering it. If you want to be in *control* while you drive, you should have two hands steering. Going faster than walking speed with only one hand on the steering wheel in a crowded place like a space station is a recipe for disaster, isn't it?

On that note, I'd probably suggest something like:
- No hands on the steering wheel: You can only go straight.
    - If you hit something solid, you will fly off.
    - If something bumps into you, they will change your direction
    - If you're going faster than base speed, you might occasionally wlose control
- One hand on the steering wheel: You get a bit of control.
    - You can steer if you're going at walking speeds.
    - If you're going faster, you don't lose control when going straight, but turning might not work perfectly.
    - If something bumps into you, you'll maintain control.
    - If you hit something, you won't fly off.
- Two hands: full control
    - You can steer at all speeds
    - You get a little spedometer or icon in your offhand, maybe?
    - You can move as fast as possible.
    - Switching from two to zero hands while moving at full speed would have some kind of consequence.

## Momentum

Here's a fun one. I know charlie's brought this up a few times. Acceleration is not just a binary thing, and if you want to get up to speed, it might be good to have to pay for it. 
It should take time to accelerate, and if you're above a certain speed when hitting walls, you should also have to pay for it damage-wise. 
Especially if you're in the sec-way. Because it's very funny to watch the warden fly head-over-heels when chasing the mime.

## Countering

So, I've already covered a few ways that these vehicles could be damaged. It's really easy to cop out and  I'm trying to think balance-wise here, which isn't really my forte, so bear with me.

I want new vehicles to be Interesting to fight, so I don't want to just make the ion rifle win against them. I think they should be able to tank a few shots, but also that taking someone out of a vehicle should be a bit more interesteing than shoot to kill and win. I would love to see something like bolas and bear traps laid out in front of an oncoming rogue ambulance, for example. 

I also think that security (or someone looking for a ride) should be able to smack someone out of a rogue vehicle with a little more than a click to unbuckle them. Sure, you should be able to shove people out of secways (and you should, because again, it's really funny), but I think the janitor should be secure aboard their noble steed. At least, unless they get batonned. Maybe holding onto the steering wheel with your hands could also keep you in a vehicle for just a little longer.

## Closing thoughts

This is partially just me trying to get some ideas out there. Reworking vehicles this heavily might be a bit of an undertaking, but I think there could be some nice baby steps once they're effectively refactored.

Iunno, I'd love to hear what people think about this. I don't know if I'll have the time before the summer to really tackle it, but I'm just happy to get this down somewhere so other people can think about it as well.

Also, there's very much the question of, who would be responsible for this? Roboticists, engineers, scientists?

~~...mechanics?~~

Once vehicles get a facelift, the foundation could certainly be laid for a new role who builds vehicles for different departments (shoutouts to Ramon for that idea). With less shitcode, and the fact that the vehicles would be built for use on-station, it could really make for something neat. That might be more of a pipe dream, though.

This might also open the door for some kind of bus driver or something, giving people a speedy tour around a large station like the farragus, possibly with a bit of skill involved in driving effectively.

Again, just food for thought.

**Comment by SpringSkipper** [comment]
something to add, is that there would probably need to be at least a slight delay to getting on or off of a vehicle to prevent gaming. This would make it awkward to abuse during escapes and better suited for what you ostensibly want vehicles to be good for: getting in and using the vehicle for a while. I don't think we want vehicles to become a major skill gap element where learning how to hotswap from driving to on-foot becomes an effective movement speed boost.

**Comment by Sayrbee** [comment]
what if we get one of those little trains for kids that you see in malls going around. I think it would be funny to be a train conductor instead of just the person at the front of the conga line
![image](https://user-images.githubusercontent.com/107023843/231330184-6977e918-8172-4e8d-9cb4-a09ccea64c46.png)
